src/compiler/utilities.ts

```diff
@@ -1071,6 +1071,15 @@ namespace ts {
         };
     }
 
+    export function createDiagnosticMessageChainFromDiagnostic(diagnostic: DiagnosticRelatedInformation): DiagnosticMessageChain {
+        return typeof diagnostic.messageText === "string" ? {

```

Comment:
I assume the true branch excludes some property of DiagnosticRelatedInformation that caused the circularity? Was it `file`?

Reply:
Yep, exactly.